src/server/semester/semester.service.ts

```diff
@@ -59,4 +72,160 @@ export class SemesterService {
+        courseIdToSpringInstance[instance.course.id] = instance;
+      });
+
+      this.logService.info('Creating the fall course instances.');
+
+      const newFallInstances = fallInstances
+        .map((fallInstance) => {
+          const springInstance: CourseInstance = courseIdToSpringInstance[
+            fallInstance.course.id
+          ];
+          return {
+            ...new CourseInstance(),
+            offered: springInstance.offered === OFFERED.RETIRED
+              ? OFFERED.RETIRED : OFFERED.BLANK,
+            course: fallInstance.course,
+            facultyCourseInstances: [],
+            meetings: [],
+          };
+        });
+
+      // Create new fall semester
+      const fallSemester = new Semester();
+      // NOTE: academic year in Semester entity is actually the calendar year
+      fallSemester.academicYear = (newAcademicYear - 1).toString();
+      fallSemester.term = TERM.FALL;
+      fallSemester.courseInstances = newFallInstances;
+
+      this.logService.info('Creating the fall non class events.');
+
+      fallSemester.nonClassEvents = existingFallSemester.nonClassEvents
+        .map((nce) => ({
+          ...new NonClassEvent(),
+          nonClassParent: nce.nonClassParent,
+          private: nce.private,
+          meetings: [],
+        }));
+
+      this.logService.info('Creating the fall absences.');
+
+      fallSemester.absences = existingFallSemester.absences
+        .map((absence) => ({
+          ...new Absence(),
+          faculty: absence.faculty,
+        }));
+
+      this.logService.info('Creating the spring course instances, non class events, and absences.');
+
+      // Create new spring semester
+      const springSemester = {
+        ...fallSemester,
+        courseInstances: fallSemester.courseInstances.map(
+          (instance) => ({ ...instance })
+        ),
+        nonClassEvents: fallSemester.nonClassEvents.map(
+          (event) => ({ ...event })
+        ),
+        absences: fallSemester.courseInstances.map(
+          (absence) => ({ ...absence })
+        ),
+        academicYear: newAcademicYear.toString(),
+        term: TERM.SPRING,
+      };
+
+      // Note that this also saves the nested entities because of
+      // the cascade set on semester
+      this.logService.info('Saving the new fall and spring semesters.');
+      await this.semesterRepository.save([fallSemester, springSemester]);
+
+      this.logService.info(`Successfully created academic year ${newAcademicYear}.`);
+    }
+    this.logService.info(`Academic year ${newAcademicYear} already exists.`);
+    return null;
+  }
+
+  public onApplicationBootstrap(): void {
+    const today = new Date();
+    if (today.getMonth() === MONTH.JUN) {
+      const yearToAdd = today.getFullYear() + 3;
+      try {
+        void this.addAcademicYear(yearToAdd);
+      } catch (e) {
+        this.logService.error(`Failed to add academic year ${yearToAdd}: ${String(e)}`);
```

Comment:
Our [`LogService#error` method](https://github.com/seas-computing/course-planner/blob/main/src/server/log/log.service.ts#L128-L148) will actually accept an `Error` as its first argument and applies some nicer builtin formatting, i.e. splitting out the message and stack trace.

So this could just be 

```ts
this.logService.error(e);
```

Reply:
Ah I see ok I updated this so that it logs the error returned ([2f02ecb](https://github.com/seas-computing/course-planner/pull/496/commits/2f02ecb09fa47763de09a54a209fa5313b26419d)).